### PR TITLE
[frugally-deep] Update to version 0.15.19-p0

### PR DESCRIPTION
**Pull request by Dobiasd**
This PR updates frugally-deep to the latest version (`0.15.19-p0`).

I have not run `./vcpkg x-add-version --all`, because I don't have a Windows machine. Is there any way for me to contribute nonetheless?

### CODE REVIEW

**Comment by Dobiasd** [comment]
@microsoft-github-policy-service agree

**Comment by mathisloge** [comment]
@Dobiasd you can run the version command on Linux and Macos, too.

**Comment by mathisloge** [comment]
You just need to call the bootstrap script beforehand

**Comment by LilyWangLL** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> After committing all other changes, the version database must be updated
> ```shell
> git add -u && git commit
> git checkout 62d01b70df227850b728f5050418b917ad6d2b32 -- versions
> ./vcpkg x-add-version --all
> ```
> 
> Diff
> You have modified or added at least one portfile where deprecated functions are used.
> 
> If you feel able to do so, please consider migrating them to the new functions: `vcpkg_install_cmake` -> `vcpkg_cmake_install` (from port `vcpkg-cmake`) `vcpkg_build_cmake` -> `vcpkg_cmake_build` (from port `vcpkg-cmake`) `vcpkg_configure_cmake` -> `vcpkg_cmake_configure` (Please remove the option `PREFER_NINJA`) (from port `vcpkg-cmake`) `vcpkg_fixup_cmake_targets` -> `vcpkg_cmake_config_fixup` (from port `vcpkg-cmake-config`) `vcpkg_extract_source_archive_ex` -> [`vcpkg_extract_source_archive`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_extract_source_archive.md) `vcpkg_build_msbuild` -> [`vcpkg_install_msbuild`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md) `vcpkg_copy_tool_dependencies` -> [`vcpkg_copy_tools`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md) `vcpkg_apply_patches` should be replaced by the `PATCHES` arguments to the "extract" helpers (e.g. [`vcpkg_from_github()`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_from_github.md))
> 
> In the ports that use the new function, you have to add the corresponding dependencies:
> 
> ```json
> {
>   "name": "vcpkg-cmake",
>   "host": true
> },
> {
>   "name": "vcpkg-cmake-config",
>   "host": true
> }
> ```
> 
> The following files are affected:
> 
> * `ports/frugally-deep/portfile.cmake`
> 
> You have modified or added at least one vcpkg.json where you should check the `license` field.
> 
> If you feel able to do so, please consider adding a "license" field to the following files:
> 
> * `ports/frugally-deep/vcpkg.json`
> 
> Valid values for the license field can be found in the [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#license)

@Dobiasd Please apply these suggestions.

**Comment by Dobiasd** [comment]
Ah, got it. Thanks. :+1: It worked, and I've committed the needed changes. :heavy_check_mark:

---

Small remark:

The [getting-started guide for Linux](https://vcpkg.io/en/getting-started.html) suggests:

```bash
https://vcpkg.io/en/getting-started.html
./vcpkg/bootstrap-vcpkg.sh
vcpkg install [packages to install]
```

It does not mention, where the `vcpkg` binary is built to, and that (at least without further copying/symlinking) the last command actually is `./vcpkg/vcpkg install [packages to install]`.

**Comment by LilyWangLL** [comment]
Please also fix the deprecated function in **portfile.cmake** and add `"license": "MIT",` in **vcpkg.json**. I have no permission to commit to your branch.

**portfile.cmake**
```
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
    -DFDEEP_BUILD_UNITTEST=OFF
    -DFDEEP_USE_TOOLCHAIN=ON
    ${FEATURE_OPTIONS}
)

vcpkg_cmake_install()
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/frugally-deep)
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")

vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```
**vcpkg.json**
```
{
  "name": "frugally-deep",
  "version-semver": "0.15.19-p0",
  "description": "Header-only library for using Keras models in C++.",
  "homepage": "https://github.com/Dobiasd/frugally-deep",
  "license": "MIT",
  "dependencies": [
    "eigen3",
    "fplus",
    "nlohmann-json",
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ],
  "features": {
    "double": {
      "description": "Use double precision"
    }
  }
}
```

**Comment by Dobiasd** [comment]
Done. :heavy_check_mark:
Thanks a lot for your detailed help! :pray:

File: ports/frugally-deep/portfile.cmake [hunk 1]
```diff
@@ -1,28 +1,14 @@
-vcpkg_from_github(
```
Comment (mathisloge) [inline ports/frugally-deep/portfile.cmake:1]: you had to leave `vcpkg_from_github` and `vcpkg_check_features` in ^^

Comment (Dobiasd) [reply ports/frugally-deep/portfile.cmake:1]: Oh, thanks! :woozy_face:

Sorry, I misunderstood Lily's comment.

**Comment by Dobiasd** [comment]
Sorry for me having no idea what I'm doing here, and thanks for all the help. :pray:

I just pushed a commit, hoping to fix the situation. But in addition, I added you two (LilyWangLL and mathisloge) as collaborators to my fork. Feel free to just push to it, if that makes things easier. :slightly_smiling_face:

File: ports/frugally-deep/portfile.cmake (hunk 1 above)
Comment (mathisloge) [reply ports/frugally-deep/portfile.cmake:1]: :D np. I'll look at this in the evening. 👍

**Review by LilyWangLL — APPROVED** [review]

**Comment by Dobiasd** [comment]
Thank you so much! :pray:

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**